src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,70 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final Integer MIN_AGE = 18;
+    private static final Integer MAX_AGE = 110;
+    private static final Integer MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkNullValues(user);
+        checkLogin(user);
+        checkAge(user);
+        checkPassword(user);
+        checkIfUserExist(user);
+        storageDao.add(user);
+        return user;
+    }
+
+    private void checkNullValues(User user) {
+        if (user == null) {
+            throw new UserDataException(UserDataException.class.getName()
+                    + " User is null");
+        }
+        if (user.getLogin() == null) {
+            throw new UserDataException(UserDataException.class.getName()
+                    + " User login is null");
+        }
+        if (user.getAge() == null) {
+            throw new UserDataException(UserDataException.class.getName()
+                    + " User age is null");
+        }
+    }
+
+    private void checkLogin(User user) {
```

Comment:
```suggestion
    private void validateLogin(User user) {
```
And let's rename all remaining methods the same way. Also pay attention to your exception messages and make them more clear